### PR TITLE
Add Project Dashboard

### PR DESCRIPTION
**Pull request by rawagner**
depends on https://github.com/openshift/console/pull/2898

screencapture-localhost-9000-k8s-cluster-projects-istio-system-dashboard-2019-10-03-13_44_08

### CODE REVIEW

File: frontend/public/components/dashboard/project-dashboard/activity-card.scss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+.co-project-dashboard__activity-body {
+  max-height: 46rem;
+}
```
Comment (vojtechszocs) [inline frontend/public/components/dashboard/project-dashboard/activity-card.scss]: Please add newline.

File: frontend/public/components/dashboard/project-dashboard/utilization-card.tsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+import { getRangeVectorStats } from '../../graphs/utils';
+import { PrometheusResponse } from '../../graphs';
+import { ProjectDashboardContext } from './project-dashboard-context';
+import { getName } from '@console/shared';
+
+export const UtilizationCard = withDashboardResources(
+  ({ watchPrometheus, stopWatchPrometheusQuery, prometheusResults }: DashboardItemProps) => {
+    const { obj } = React.useContext(ProjectDashboardContext);
+    const projectName = getName(obj);
+    const cpuQuery = `namespace:container_cpu_usage:sum{namespace='${projectName}'}[60m:5m]`;
```
Comment (vojtechszocs) [inline frontend/public/components/dashboard/project-dashboard/utilization-card.tsx]: Please extract queries based on current convention, i.e. `./queries.ts`.

**Comment by vojtechszocs** [comment]
Will rebase again once the pre-requisite #2898 is merged.

**Comment by vojtechszocs** [comment]
@spadgett @jelkosz I've rebased the PR without any further changes, solving merge conflicts only.

Suggestions in my https://github.com/openshift/console/pull/2895#pullrequestreview-298264798 can be done as follow-up. Is it OK to give LGTM on this PR?

**Comment by jelkosz** [comment]
@vojtechszocs yeah, lets get this huge PR in before it will get conflicts again and solve the issues in followup

**Comment by vojtechszocs** [comment]
#2898 is merged now, needs one more rebase.

**Comment by vojtechszocs** [comment]
/lgtm

Adding LGTM based on https://github.com/openshift/console/pull/2895#issuecomment-539519282.

**Comment by spadgett** [comment]
/hold

This is failing e2e-aws-console

**Comment by vojtechszocs** [comment]
> This is failing e2e-aws-console

CI error indicates a problem with `integration-tests/tests/performance.scenario.ts` failing to download webpack bundle for route `persistent-volume-claim`, checking.

**Comment by vojtechszocs** [comment]
LGTM on https://github.com/openshift/console/pull/2895/commits/85881469fbccce04a2145aaf0a47395e126fdafe

**Comment by vojtechszocs** [comment]
Updated Sam's commit, `ci/prow/frontend` should pass now.

**Comment by vojtechszocs** [comment]
/hold cancel

`e2e` test is now passing. Thank you @spadgett for the help!

**Comment by vojtechszocs** [comment]
Rebased & squashed commits into one.

File: frontend/public/components/dashboard/project-dashboard/activity-card.tsx [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+        };
+      }
+    }, [watchK8sResource, stopWatchK8sResource, projectName]);
+    return <RecentEventsBody events={resources.events as FirehoseResult<EventKind[]>} />;
+  },
+);
+
+const getResourceActivities = (flags: FlagsObject, k8sModels: ImmutableMap<string, K8sKind>) =>
+  plugins.registry.getDashboardsOverviewResourceActivities().filter((e) => {
+    const model = k8sModels.get(e.properties.k8sResource.kind);
```
[the remark this answers is not in the document]
Comment (rawagner) [reply frontend/public/components/dashboard/project-dashboard/activity-card.tsx]: The idea here is that we reuse Overview Dashboard Activity extension point for Project dashboard too but we will filter the resources which are namespaced

**Comment by spadgett** [comment]
@rawagner This doesn't add any of the console project links to the dashboard. Is that on your radar?

cc @rhamilto

**Comment by spadgett** [comment]
I opened this as a Bugzilla since it needs to be tracked as a release blocker.

https://bugzilla.redhat.com/show_bug.cgi?id=1763836

**Comment by rawagner** [comment]
> @rawagner This doesn't add any of the console project links to the dashboard. Is that on your radar?
> 
> cc @rhamilto

it is, i will be looking at it this week. Thanks for creating bugzilla

**Pull request merged**